providers/firestore.js

```diff
@@ -0,0 +1,183 @@
+		return await this.db.collection(table).doc(id).set(this.parseUpdateInput(doc));
+	}
+
+	/**
+	 * Updates the values of a Document/Field
+	 * @param {string} table Name of the Collection that contains the Document
+	 * @param {string} id  Name of the Document/field
+	 * @param {(ConfigurationUpdateResultEntry[] | [string, any][] | Object<string, *>)} doc the object you want to insert in the table.
+	 * @returns {Promise<WriteResult>}
+	 */
+	async update(table, id, doc) {
+		return await this.db.collection(table).doc(id).set(this.parseUpdateInput(doc), { merge: true });
+	}
+
+	/**
+	 * Deleted a particular Document/Field from the Collection/Table
+	 * @param {string} table Name of the collection/table
+	 * @param {string} id Name of the Document/Key to delete
+	 * @returns {Promise<WriteResult>}
+	 */
+	async delete(table, id) {
+		return await this.db.collection(table).doc(id).delete();
+	}
+	/**
+	 * Replaces a Document with a new Document specified by the user *
+	 * @param {string} table Name of the Collection
+	 * @param {Object} id The Filter used to select the document to update
+	 * @param {Object} [doc={}] The Document that replaces the matching document
+	 * @returns {Promise}
+	 */
+	replace(...args) {
+		return this.create(...args);
+	}
+
+	/**
+	 * Update or insert a new value to all entries
+	 * @param {string} table Name of the Collectiom
+	 * @param {string} path The Field to be updated in al the Documents
+	 * @param {*} newValue The value to to updated in all the Documents
+	 */
+	async updateValue(table, path, newValue) {
+		const keys = await this.getKeys(table);
+
+		if (typeof path === 'object' && typeof newValue === 'undefined') {
+			await Promise.all(keys.map(doc => this.update(table, doc, path)));
+		}
+
+		if (typeof path === 'string' && typeof newValue !== 'undefined') {
+			await Promise.all(keys.map(doc => this.update(table, doc, { path: newValue })));
+		}
+
+		throw new TypeError(`Expected an object as first parameter or a string and a non-undefined value. Got: ${typeof key} and ${typeof value}`);
+	}
+
+	/**
+	 * Remove a value or object from all entries.
+	 * @param {string} table Name of the Collection
+	 * @param {string} path The Field to be updated in al the Documents
+	 * @param {*} newValue The value to to updated in all the Documents
+	 */
+	async removeValue(table, path) {
+		const keys = await this.getKeys(table);
+		if (typeof path === 'object' && typeof newValue === 'undefined') {
+			for (const key of Object.keys(path)) path[key] = FieldValue.deleteValue();
+			await Promise.all(keys.map(doc => this.update(table, doc, path)));
+		}
+
+		if (typeof path === 'string' && typeof newValue !== 'undefined') {
+			await Promise.all(keys.map(doc => this.update(table, doc, { path: FieldValue.deleteValue() })));
+		}
+
+		throw new TypeError(`Expected an object as first parameter or a string and a non-undefined value. Got: ${typeof key} and ${typeof value}`);
+	}
+	set(...args) {
```

Comment:
set and insert are now removed from all providers to prevent confusion of the standard method (`create()`).